src/nile/utils/status.py

```diff
@@ -0,0 +1,100 @@
+"""Functions used to find/track/debug a transaction status."""
+
+import json
+import logging
+import subprocess
+import time
+from collections import namedtuple
+from enum import Enum
+
+from nile.common import RETRY_AFTER_SECONDS, get_network_parameter_or_set_env
+from nile.utils.debug import debug_message
+
+TransactionStatus = namedtuple(
+    "TransactionStatus", ["tx_hash", "status", "error_message"]
+)
+
+
+def status(
+    tx_hash, network, status_type=None, contracts_file=None
+) -> TransactionStatus:
+    """Fetch a transaction status.
+
+    Optionally track until resolved (accepted on L2 or rejected) and/or
+    use available contracts to help locate the error. Debug implies track.
+    """
+    command = ["starknet", "tx_status", "--hash", hex(tx_hash)]
+    command += get_network_parameter_or_set_env(network)
+
+    logging.info("⏳ Querying the network for transaction status...")
+
+    receipt = _get_tx_receipt(tx_hash, command, status_type)
+
+    if not receipt.status.is_rejected:
+        return TransactionStatus(tx_hash, receipt.status, None)
+
+    error_message = receipt.receipt["tx_failure_reason"]["error_message"]
+    if status_type == "debug":
```

Comment:
any reason for this not to be the default? when wouldn't you want `debug` enabled?

Reply:
In my opinion, I don't want my terminal to be blocked for a couple of minutes. I usually want to send the transaction and check it later, not block the terminal where I'm working (I could open more terminals, but this is not what I expect as a regular flow as a developer).

Reply:
I considered setting it as default but reconsidered with the same point @ericnordelo expressed^. I think opt-in seems more appropriate...perhaps, we could include an `auto-debug` feature in the future?